### PR TITLE
Add an endpoint for actuator health

### PR DESCRIPTION
**Pull request by dbraquart**
This new endpoint uses spring boot Actuator (/health part) to retrieve the list of all available optional services for study-app. These services are defined in application.yaml: now the gridsuite services are defined as a list, where we can put an optional parameter "optional: true".

We can test this endpoint using swagger (_http://localhost:5001/webjars/swagger-ui/index.html?configUrl=/v3/api-docs/swagger-config#/Study%20server/getOptionalUpServices_).

This PR has been also tested wih Minikube, and [gateway PR](https://github.com/gridsuite/gateway/pull/86) and [GridStudy PR](https://github.com/gridsuite/gridstudy-app/pull/1419)

### CODE REVIEW

**Review by Walid-Sahnoun — APPROVED** [review]
code ok 
test ok

**Review by ayolab — APPROVED** [review]

File: src/main/java/org/gridsuite/study/server/service/ActuatorHealthService.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        try {
+            List<CompletableFuture<String>> listOfFutures = optionalServices.keySet().stream().map(this::isUpFuture).toList();
+            CompletableFuture<List<String>> futureOfList = CompletableFuture
+                    .allOf(listOfFutures.toArray(new CompletableFuture[0]))
+                    .thenApply(v -> listOfFutures.stream().map(CompletableFuture::join).filter(Objects::nonNull).toList());
+            return futureOfList.get();
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            return List.of();
+        } catch (Exception e) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/service/ActuatorHealthService.java:84]: Change by `ExecutionException`

File: src/main/java/org/gridsuite/study/server/service/ActuatorHealthService.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            return List.of();
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/service/ActuatorHealthService.java:85]: Log the error

File: src/main/java/org/gridsuite/study/server/service/ActuatorHealthService.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        } catch (RestClientException e) {
+            return false;
+        }
+        try {
+            JsonNode node = OBJECT_MAPPER.readTree(result).path(ACTUATOR_HEALTH_STATUS_JSON_FIELD);
+            if (!node.isMissingNode() && node.asText().equalsIgnoreCase(ACTUATOR_HEALTH_STATUS_UP)) {
+                return true;
+            }
+        } catch (JsonProcessingException e) {
+            return false;
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/service/ActuatorHealthService.java:108]: Log the error

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/study/server/utils/StudyUtils.java [hunk 4]
```diff
@@ -48,4 +51,13 @@ private static String parseHttpError(String responseBody) {
 
         return responseBody;
     }
+
+    public static String getServiceUri(RemoteServicesProperties properties, String serviceName) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/utils/StudyUtils.java:55]: Move it in `RemoteServicesProperties` class (instance method)

File: src/test/java/org/gridsuite/study/server/ActuatorHealthTest.java [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+ * @author David Braquart <david.braquart at rte-france.com>
+ */
+
+@RunWith(SpringRunner.class)
+@AutoConfigureMockMvc
+@DisableElasticsearch
+@SpringBootTest
+public class ActuatorHealthTest {
+
+    private WireMockServer wireMockServer;
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/study/server/ActuatorHealthTest.java:46]: Separate attibutes by newline ?

File: src/test/java/org/gridsuite/study/server/ActuatorHealthTest.java [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+
+    private List<String> getOptionalServicesNames() {
+        return remoteServicesProperties.getServices().stream()
+                .filter(RemoteServicesProperties.Service::getOptional)
+                .map(RemoteServicesProperties.Service::getName)
+                .toList();
+    }
+
+    @Test
+    public void testActuatorHealthUp() throws Exception {
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/study/server/ActuatorHealthTest.java:72]: Add a test with only some optional services ?

File: src/test/java/org/gridsuite/study/server/service/client/dynamicmapping/DynamicMappingClientTest.java [hunk 7]
```diff
@@ -68,7 +69,8 @@ public void setup() {
         super.setup();
 
         // config client
-        dynamicMappingClient = new DynamicMappingClientImpl(initMockWebServer(), restTemplate);
+        dynamicMappingClient = new DynamicMappingClientImpl(null, restTemplate);
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/study/server/service/client/dynamicmapping/DynamicMappingClientTest.java:72]: @Autowired
    private RemoteServicesProperties remoteServicesProperties;

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/study/server/service/ActionsService.java [hunk 8]
```diff
@@ -59,7 +58,7 @@ public Integer getContingencyCount(UUID networkUuid, String variantId, List<Stri
                     new ParameterizedTypeReference<List<Contingency>>() {
                     }).getBody();
 
-            return contingencies.size();
+            return contingencies == null ? 0 : contingencies.size();
```
[the remark this answers is not in the document]
Comment (dbraquart) [reply src/main/java/org/gridsuite/study/server/service/ActionsService.java:61]: this avoid a potential NPE

File: src/test/java/org/gridsuite/study/server/ActuatorHealthTest.java (hunk 6 above)
Comment (dbraquart) [reply src/test/java/org/gridsuite/study/server/ActuatorHealthTest.java:76]: too hard to do (need multiple wiremock server ?)

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**